modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/NoLossBurstTopicMessageDeliveryImpl.java

```diff
@@ -111,11 +115,11 @@ public int deliverMessageToSubscriptions(StorageQueue storageQueue) throws
 
                 message.markAsScheduledToDeliver(subscriptionsToDeliver);
 
+                iterator.remove();
```

Comment:
That was to prevent messages being removed from both readButUndelivered list and the deliveredButUnacked map